enacts/calc.py

```diff
@@ -332,6 +332,38 @@ def onset_date(
     return onset_delta
 
 
+def cess_date_step(cess_yesterday, spell_length):
+    """Updates cessation date delta according to today's soil moisture spell length
+
+    A cessation date is found at the first day of the first dry spell.
+    Today's cessation date delta is one day further away than yesterday's.
+
+    Parameters
+    ----------
+    cess_yesterday : DataArray[np.timedelta64]
+        Yesterday's distance in days from cessation date.
+        Can not have a time dimension of size greater than 1.
+    spell_length : DataArray[np.timedelta64]
+        Today's length in days of a dry spell.
+    
+    Returns
+    -------
+        DataArray[np.timedelta64]
+        Updated cessation date delta from today.
+
+    Notes
+    -----
+        It is understood that a spell is defined once long enough.
+        Thus its length is defined as soon as the spell reaches its minimum length.
+        Thus today's spell length if NaT if no spell has been long enough so far.
+        Once a cessation date is found (hence its delta from today),
+        the value of `spell_length` is ignored.
+    """
+    return cess_yesterday.where(
+        ~np.isnat(cess_yesterday), other=(np.timedelta64(2, "D") - spell_length)
+    ) - np.timedelta64(1, "D")
```

Comment:
```
(2 - spell_length) - 1 =
1 - spell_length =
- (spell_length - 1)
```
That means the day after the dry spell started, doesn't it? Is that a mistake? Doesn't seem right but maybe I'm not interpreting it the way you intended.

It might be a little easier to reason about if you made cess_delta positive (and then subtract it instead of adding it when it comes time to convert it to a date).

Alternatively... I notice you're not using the time coordinate, unlike in the water balance step function. What if you make T represent the date you're currently processing, either as an absolute date or a timedelta since the start of the search period, and when you find a dry spell, put the T of the start of the dry spell into cess_delta. Then cess_delta represents the difference between the cessation date and the start of the search period, rather than between "now" and the cessation date. Seems like it might be simpler than incrementing every delta on each implementation, and maybe more efficient too.

Reply:
Let's say the spell length that makes a cessation is 5. Somewhere today spell length reaches 5 and -(5-1) is -4. 4 days from today (the 5th day of the 5-day long spell) was the 1st day of the 5-day long spell.

I believe (nearly) all you are saying otherwise will happen in the looping function, not in the step one. water_balance_step actually ignores the time dimension too. It's the loop calling it that involves it and that manipulates it the way that makes best sense for it.

So in that regards, I do expect the loop to eventually reset the reference time for those delta to the start of the search (if we keep doing in that way -- and btw, that will "flip" back the deltas to be positive values). The loop will get a new "now" time dim at each iteration, also needed by the water_balance_step (for whom the start of search date doesn't have meaning btw). Having both a now T and another reference T I feel will expose to complications as I will have to make sure operations won't try to broadcast one with the other, and fail because they are not the same value, so I will have to constantly remove one and put it back or something like that.

All the above is more gut feeling than careful assessment though. I would assume you are correct about the efficiency since there won't be an addition done at each iteration but only assignments when appropriate.

What do you think?

Reply:
OK for the meaning of `spell_length - 1`. Any thoughts about using a positive number, i.e. making the negative implicit? (It's not possible for a dry spell to start in the future and continue through today.)

I don't entirely follow your argument about the different ways of handling time, but it does seem natural for the step function not to know anything about the time dimension.

Reply:
We can make the negative implicit and operate in the positive numbers by saying that the output of cess_date_step is a distance (delta) such as cess_date = today - delta (currently it is assuming to be today + delta). That means that in the loop function, when the loop is over, we'll have a positive delta with respect to last T of the loop (T_last) so that cess_date = T_last - delta. When we give it the coordinate T as the start of the search of the cessation date, the new_delta will become T_last - delta - T_start_of_search, making a positive new_delta again, but then, the cess_date will be
T_start_of_search + new_delta, being a bit incoherent with the behavior of cess_date_step... but cess_date_step is of different nature of cess_date so... either way is fine with me.

Reply:
I could write the loop function in this PR if it helps see the whole thing together.

Reply:
The sign of the delta is just an aesthetic preference, and I don't have the full picture in my head yet, so it's fine to leave it if you don't share the preference.

Seeing the whole loop would probably help.